webapp/src/workspace.ts

```diff
@@ -607,6 +609,14 @@ export function syncAsync(): Promise<pxt.editor.EditorSyncState> {
     checkSession();
 
     return impl.listAsync()
+        .catch((e) => {
+            // There might be a problem with the native databases. Switch to memory for this session so the user can at
```

Comment:
this kind of crashes can happen at any time and this might be too late now? is listAsync() called on load only?

Reply:
It's called on init, and when going back to home page

I can add a flag `isInit` and only do the workspace switch if we're during initialization

Reply:
Yes then your fix is good. We should add a notification banner in the future to explain what's going on.